inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -183,12 +183,16 @@ def _is_arxiv_url(url):
         source='arXiv' if form_fields.get('categories') else 'CrossRef'
     )
 
-    if not _is_arxiv_url(form_fields.get('url', '')):
-        builder.add_url(url=form_fields.get('url'))
-        obj.extra_data['submission_pdf'] = form_fields.get('url')
-
-    if not _is_arxiv_url(form_fields.get('url', '')):
-        builder.add_url(url=form_fields.get('additional_url'))
+    form_url = form_fields.get('url')
+    form_additional_url = form_fields.get('additional_url')
+    if form_url and not _is_arxiv_url(form_url):
+        obj.extra_data['submission_pdf'] = form_url
+        if not form_additional_url:
+            builder.add_url(url=form_url)
+
+    if form_additional_url and not _is_arxiv_url(form_additional_url):
+        if form_url != form_additional_url:
```

Comment:
Actually this should not be possible since one field ensures that it is a PDF and the other ensures that it is not a PDF. So if validators work properly this cannot happen.